app/src/main/java/kth/jjve/moveable/MainActivity.java

```diff
@@ -427,45 +432,49 @@ public void savingCancelled() {
     @Override
     public void onSensorChanged(SensorEvent event) {
         if (event.sensor.getType()==Sensor.TYPE_ACCELEROMETER){
-            ax=event.values[0];
-            ay=event.values[1];
-            az=event.values[2];
+            float ax=event.values[0];
+            float ay=event.values[1];
+            float az=event.values[2];
 
             // rotation from acceleration data
-            roll_x = Math.toDegrees(Math.atan(ay / (Math.sqrt(Math.pow(ax, 2) + Math.pow(ax, 2)))));
-            pitch_y = Math.toDegrees(Math.atan(ax / (Math.sqrt(Math.pow(ay, 2) + Math.pow(ax, 2)))));
-            yaw_z = Math.toDegrees(Math.atan( (Math.sqrt(Math.pow(ax, 2) + Math.pow(ay, 2)) / az )));
+            pitchRollYaw = DataProcess.rotFromAcc(ax, ay, az);
         }
-        //Log.i(LOG_TAG, "Acceleration    x: " + ax + ", y: " + ay + ", z: " + az);
-        //Log.i(LOG_TAG, "roll_x: " + roll_x + ", pitch_y: " + pitch_y + ", yaw_z: " + yaw_z);
+        tempIntRotAcc.setText("Rotation from accelerometer: " + Math.round(pitchRollYaw[0]) + ", " + Math.round(pitchRollYaw[1]) + ", " + Math.round(pitchRollYaw[2]));
+
+        EWMA_filtered_values[0] = DataProcess.EMWA_filter(pitchRollYaw[0], (float) 0.5, EWMA_filtered_values[0]);
+        EWMA_filtered_values[1] = DataProcess.EMWA_filter(pitchRollYaw[1], (float) 0.5, EWMA_filtered_values[1]);
+        EWMA_filtered_values[2] = DataProcess.EMWA_filter(pitchRollYaw[2], (float) 0.5, EWMA_filtered_values[2]);
+        Log.i(LOG_TAG, "EMWA filter: " + EWMA_filtered_values[0] + ", " + EWMA_filtered_values[1] + ", " + EWMA_filtered_values[2]);
 
         if (event.sensor.getType()==Sensor.TYPE_GYROSCOPE){
-            gx=event.values[0];
-            gy=event.values[1];
-            gz=event.values[2];
+            float gx = (float) Math.toDegrees(event.values[0]);
+            float gy= (float) Math.toDegrees(event.values[1]);
```

Comment:
Maybe you can make these functions void and store gy locally in the process class. Makes life a bit easier

Reply:
not sure I get what you're meaning, but I will look into it